nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/scroll/RepositoryScroll.java

```diff
@@ -117,6 +122,19 @@ public void close() {
         }
     }
 
+    /**
+     * Helper to load a list of documents. Documents without read access or that does not exist are not returned.
+     * @param session the Core session
+     * @param documentIds the document Ids' list.
+     * @return the corresponding DocumentModelList.
+     */
+    public static DocumentModelList loadDocuments(CoreSession session, List<String> documentIds) {
+        if (documentIds == null || documentIds.isEmpty()) {
+            return new DocumentModelListImpl(0);
+        }
+        return session.query(String.format(SELECT_DOCUMENTS_IN, String.join("', '", documentIds)));
+    }
```

Comment:
see my comment on https://github.com/nuxeo/nuxeo/pull/3693/commits/48c62d0e6e904561781b271ae17ba1c68f67712e#r370538308
furthermore i think you cas use `StaticScroll` as you have the list of ids .... you can ask Benoit for more context

and for more context, you can rewrite this method as below:
````
  String inClause = documentIds.stream().collect(Collectors.joining(", ", "'", "'"));
        String query = String.format(SELECT_DOCUMENTS_IN, NXQL.ECM_UUID, inClause);
        return session.query(query);
```
where ` SELECT_DOCUMENTS_IN = "SELECT * FROM Document, Relation WHERE %s IN (%s)";`